core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -506,47 +520,53 @@ class CtxSettings : public SettingsBase {
         }
     }
 
-    [[nodiscard]] std::optional<SettingsCtx> activateContext(SettingsCtx ctx = {}) override {
+    [[nodiscard]] NO_INLINE std::optional<SettingsCtx> activateContext(SettingsCtx ctx = {}) override {
         if (ctx.time == 0ULL) {
             ctx.time = settings::convertTimePointToUint64Ns(std::chrono::system_clock::now());
 #ifdef __EMSCRIPTEN__
             ctx.time += _timePrecisionTolerance;
 #endif
         }
 
-        const auto bestMatchSettingsCtx = findBestMatchSettingsCtx(ctx);
-        if (bestMatchSettingsCtx != std::nullopt) {
-            if (bestMatchSettingsCtx == _activeCtx) {
-                // context and time are the same -> do not change stagedParameters
-                return bestMatchSettingsCtx;
-            } else if (bestMatchSettingsCtx.value().context == _activeCtx.context) {
-                // context is the same but time is different -> find the bestMatchParameters, add only (isWritable and not in autoUpdate) to stagedParameters
-                auto parameters = getBestMatchStoredParameters(ctx);
-                if (parameters != std::nullopt) {
-                    const auto   currentAutoUpdateParameters = _autoUpdateParameters.at(bestMatchSettingsCtx.value());
-                    auto         notAutoUpdateView           = parameters.value() | std::views::filter([&currentAutoUpdateParameters](const auto& pair) { return !currentAutoUpdateParameters.contains(pair.first); });
-                    property_map notAutoUpdateParameters(notAutoUpdateView.begin(), notAutoUpdateView.end());
-
-                    std::ignore = setStagedImpl(notAutoUpdateParameters);
-                    _activeCtx  = bestMatchSettingsCtx.value();
-                    setChanged(true);
+        const std::optional<SettingsCtx> bestMatchSettingsCtx = findBestMatchSettingsCtx(ctx);
+        if (!bestMatchSettingsCtx || bestMatchSettingsCtx == _activeCtx) {
+            return bestMatchSettingsCtx;
+        }
+
+        if (bestMatchSettingsCtx.value().context == _activeCtx.context) {
+            std::optional<property_map> parameters = getBestMatchStoredParameters(ctx);
+            if (parameters) {
+                const std::set<std::string>& currentAutoUpdateParams = _autoUpdateParameters.at(bestMatchSettingsCtx.value());
+                // auto                         notAutoUpdateView       = parameters.value() | std::views::filter([&](const auto& pair) { return !currentAutoUpdateParams.contains(pair.first); });
+                // property_map                 notAutoUpdateParams(notAutoUpdateView.begin(), notAutoUpdateView.end());
+
+                // the following is more compile-time friendly
+                property_map notAutoUpdateParams;
+                for (const auto& pair : parameters.value()) {
+                    if (!currentAutoUpdateParams.contains(pair.first)) {
+                        notAutoUpdateParams.insert(pair);
+                    }
                 }
```

Comment:
I'm not sure about the gain of compilation time, but we might still need to consider using ranges, where possible?

Reply:
re gain: see numbers above. Unfortunately quite significant and not in favour of ranges. :-|